catalyst-gateway-crates/c509-certificate/src/c509_extensions/mod.rs

```diff
@@ -0,0 +1,189 @@
+//! C509 Extension as a part of TBSCertificate used in C509 Certificate.
+//! Extension fallback of C509 OID extension
+//! Given OID if not found in the registered OID table, it will be encoded as a PEN OID.
+//! If the OID is not a PEN OID, it will be encoded as an unwrapped OID.
+//!
+//! Extensions and Extension can be encoded as the following:
+//! Extensions = [ * Extension ] / int
+//! Extension = ( extensionID: int, extensionValue: any ) //
+//! ( extensionID: ~oid, ? critical: true,
+//!   extensionValue: bytes ) //
+//! ( extensionID: pen, ? critical: true,
+//!   extensionValue: bytes )
+//!
+//! For more information about Extensions, visit [C509](https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/)
+
+pub mod alt_name;
+pub mod extension;
+
+use std::fmt::Debug;
+
+use asn1_rs::{oid, Oid};
+use extension::{Extension, ExtensionValue};
+use minicbor::{encode::Write, Decode, Decoder, Encode, Encoder};
+
+/// OID of KeyUsage extension
+static KEY_USAGE_OID: Oid<'static> = oid!(2.5.29 .15);
+
+/// A struct of C509 Extensions
+///
+/// # Fields
+/// * `extensions` - A vector of `Extension`.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Extensions {
+    extensions: Vec<Extension>,
+}
```

Comment:
```suggestion
pub struct Extensions(Vec<Extension>);
```

Reply:
I personally don't like the `self.0` syntax and in case we need to add more stuff to Extensions, this might be easier way to handle it